crates/ika-core/src/dwallet_mpc/crytographic_computation/mpc_computations/dwallet_dkg.rs

```diff
@@ -106,6 +191,128 @@ impl DWalletDKGAdvanceRequestByCurve {
     }
 }
 
+#[derive(Clone, Debug, Eq, PartialEq, strum_macros::Display)]
+pub enum DWalletImportedKeyVerificationPublicInputByCurve {
+    #[strum(to_string = "dWallet Imported Key Verification Public Input for curve Secp256k1")]
+    Secp256K1DWalletImportedKeyVerification(
+        <Secp256K1DWalletImportedKeyVerificationParty as Party>::PublicInput,
+    ),
+    #[strum(to_string = "dWallet Imported Key Verification Public Input for curve Secp256r1")]
+    Secp256R1DWalletImportedKeyVerification(
+        <Secp256R1DWalletImportedKeyVerificationParty as Party>::PublicInput,
+    ),
+    #[strum(to_string = "dWallet Imported Key Verification Public Input for curve Curve25519")]
+    Curve25519DWalletImportedKeyVerification(
+        <Curve25519DWalletImportedKeyVerificationParty as Party>::PublicInput,
+    ),
+    #[strum(to_string = "dWallet Imported Key Verification Public Input for curve Ristretto")]
+    RistrettoDWalletImportedKeyVerification(
+        <RistrettoDWalletImportedKeyVerificationParty as Party>::PublicInput,
+    ),
+}
+
+impl DWalletImportedKeyVerificationPublicInputByCurve {
+    pub fn try_new(
+        session_identifier: CommitmentSizedNumber,
+        curve: &DWalletCurve,
+        encryption_key_public_data: &VersionedNetworkEncryptionKeyPublicData,
+        centralized_party_message: &[u8],
+    ) -> DwalletMPCResult<Self> {
+        let public_input = match curve {
+            DWalletCurve::Secp256k1 => {
+                let protocol_public_parameters =
+                    encryption_key_public_data.secp256k1_protocol_public_parameters();
+                let centralized_party_message: VersionedImportedDWalletPublicOutput =
```

Comment:
why is this `VersionedImportedDWalletPublicOutput`? Should be `VersionedImportedDWalletCentralizedPartyMessage` or something